highway.tex

```diff
@@ -24,807 +26,1323 @@
+%
+The above strategy requires, in such a case, sending endorsements only for units created in these few rounds that happen after round $r$.
+%
+The honest units that are created later, never cite any "new" units by $W$ and thus do not require endorsements.
 
-%\subsubsection{Ensuring Validators are Up to Date}
+\subsection{Weighted Consensus}
 
-%In order to ensure that none of the duplicate chains of votes produced by an equivocation remain active for very long, it is important to keep validators up to date, and in particular ensure that if any validator $W$ is known to be an equivocator by any honest validator that all other validators will be forced to acknowledge this fact (and thus by Criteria 2 stop naively citing new messages by $W$) after a small number of rounds. In order to enforce this, we introduce one final validity criteria:
-%\paragraph{Criteria 4:} A vote $\mu$ sent by a validator $V$ satisfies Criteria 4 if the following holds. Let $\mu'$ be the $N^{th}$-to-last message sent by $V$. Let $\mathcal{C}$ be the set of $V$ along with other validators $W \not\in E(\mu)$ so that if $\lambda$ was the most recently sent message by $W$ in $\mu$'s justification, then the time that $\mu'$ was sent was no later than $N$ of $\lambda$'s round lengths after $\lambda$. Then $\mu$ satisfies Criteria 4 if and only if the weight of the validators in $\mathcal{C}$ is at least $1/3$ of the total validator weight.
+So far it has been assumed that the opinion of every validator is equally important in the process of achieving consensus.
+In this subsection, we describe the modifications allowing the Highway protocol to be run in a scenario where each validator has an associated integer weight, corresponding to its voting power -- a useful version for example when constructing a proof-of-stake blockchain. 
 
-%Assuming that the honest validators comprise at least $2/3$rds of the total validator weight, it is clear that any message satisfying Criteria 4 contains in its justification a relatively recent message sent by each validator in $\mathcal{C}$, which by the weight restriction must contain at least one honest validator. Since any honest validator learning about an equivocation will quickly (assuming network stability) be communicated to other honest validators, this means that if messages are being delivered on time, then any vote satisfying Criteria 4 must acknowledge an equivocating validator not too long after the first honest validator becomes aware of such an equivocation. This along with Criteria 2 ensures that these duplicate chains will not persist for very long.
+Let the sum of all validator weights equals $N$, and the function $\ww(V)$ denotes the weight associated with validator $V$.
+In such version, the function $\mathrm{total}$ being the basis for the GHOST rule has to be replaced with the one counting the \emph{total weight} of validators supporting a given option, instead of their number. 
```

Comment:
```suggestion
In such a version, the function $\mathrm{total}$ being the basis for the GHOST rule has to be replaced with the one counting the \emph{total weight} of validators supporting a given option, instead of their number. 
```